src/simulators/tensor_network/tensor.hpp

```diff
@@ -0,0 +1,277 @@
+/**
+ * This code is part of Qiskit.
+ *
+ * (C) Copyright IBM 2018, 2019, 2022.
+ *
+ * This code is licensed under the Apache License, Version 2.0. You may
+ * obtain a copy of this license in the LICENSE.txt file in the root directory
+ * of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Any modifications or derivative works of this code must retain this
+ * copyright notice, and modified files need to carry a notice indicating
+ * that they have been altered from the originals.
+ */
+
+
+
+#ifndef _tensor_hpp_
+#define _tensor_hpp_
+
+
+#include <complex>
+#include <vector>
+
+#include "framework/types.hpp"
+#include "framework/utils.hpp"
+
+
+namespace AER {
+namespace TensorNetwork {
+
+
+template <typename data_t>
+class Tensor
+{
+protected:
+  int base_;
+  int rank_;
+  std::vector<int64_t> extents_;
+  int64_t size_;
+  std::vector<std::complex<data_t>> tensor_;
+  std::vector<int32_t> modes_;
+  reg_t qubits_;
+  bool sp_tensor_;
```

Comment:
could you leave some comments on each member? Maybe `tensor_` is data of multi-dimensional array defined `rank_` and `modes_`.
Also, why `int64_t` and `int32_t` are used here? Can they be replaced with `uint64_t` and `uint32_t`? Or can they be negative?

Reply:
comment added

Integer types defined in here is supported types in cuTensorNet, the same type is used to avoid data conversion.